store/customPreset.js

```diff
@@ -0,0 +1,86 @@
+import { groupRepeatedUnits, createRepetition } from '../utils/helpers'
+
+export const state = () => ({
+  columns: 4,
+  rows: 2,
+  colArr: [],
+  rowArr: [],
+  childarea: [],
+  previewarea: null
+})
+
+export const getters = {
+  colTemplate(state) {
+    const unitGroups = groupRepeatedUnits(state.colArr)
+
+    return createRepetition(unitGroups)
+  },
+  rowTemplate(state) {
+    const unitGroups = groupRepeatedUnits(state.rowArr)
+    return createRepetition(unitGroups)
+  },
+  divNum(state) {
+    return state.columns * state.rows
+  },
+  rowNumber(state) {
```

Comment:
consider return single item from the state as useless.
use `getters` for calculations, filtering, sorting, etc